app/lib/actions/katello/repository/update_content_urls.rb

```diff
@@ -0,0 +1,19 @@
+module Actions
+  module Katello
+    module Repository
+      class UpdateContentUrls < Actions::EntryAction
+        def plan(content_to_update)
+          concurrence do
+            content_to_update.each do |content|
+              content.root_repositories.each do |root|
+                root.repositories.each do |repo|
```

Comment:
I think there could be a decent refactor here. We could presumably give multiple contents with  the same root repository and the way the code is written would result in ::UpdateRedhatRepository being called for a given repo multiple times. Would it be a good idea to collect the unique set of root_repositories and call plan_action for all of their repos?

Reply:
ie
```
content_to_update.flat_map(&:repositories).each do |repo|
  plan_action(Katello::Repository::UpdateRedhatRepository, repo)
end
```

Reply:
per our convo and  based on the content.rb#repositories method this might be the simplest (should be the same exact result)

```
content_to_update.each do |content|
  content.repositories.each do |repo|
     plan_action ...
  end
end
```